chapter_preliminaries/calculus.ipynb

```diff
@@ -0,0 +1,465 @@
+       "var trace0 =\n",
+       "{\n",
+       "x: [\"0.0\",\"0.1\",\"0.2\",\"0.30000000000000004\",\"0.4\",\"0.5\",\"0.6000000000000001\",\"0.7000000000000001\",\"0.8\",\"0.9\",\"1.0\",\"1.1\",\"1.2000000000000002\",\"1.3\",\"1.4000000000000001\",\"1.5\",\"1.6\",\"1.7000000000000002\",\"1.8\",\"1.9000000000000001\",\"2.0\",\"2.1\",\"2.2\",\"2.3000000000000003\",\"2.4000000000000004\",\"2.5\",\"2.6\",\"2.7\",\"2.8000000000000003\",\"2.9000000000000004\"],\n",
+       "y: [\"0.0\",\"-0.37\",\"-0.68\",\"-0.9300000000000002\",\"-1.12\",\"-1.25\",\"-1.32\",\"-1.33\",\"-1.2799999999999998\",\"-1.17\",\"-1.0\",\"-0.7699999999999996\",\"-0.47999999999999954\",\"-0.1299999999999999\",\"0.28000000000000025\",\"0.75\",\"1.2800000000000011\",\"1.870000000000001\",\"2.5200000000000005\",\"3.2300000000000013\",\"4.0\",\"4.83\",\"5.720000000000002\",\"6.670000000000002\",\"7.680000000000003\",\"8.75\",\"9.88\",\"11.070000000000004\",\"12.320000000000002\",\"13.630000000000003\"],\n",
+       "mode: 'lines',\n",
+       "xaxis: 'x',\n",
+       "yaxis: 'y',\n",
+       "type: 'scatter',\n",
+       "name: 'f(x)',\n",
+       "};\n",
+       "var trace1 =\n",
+       "{\n",
+       "x: [\"0.0\",\"0.1\",\"0.2\",\"0.30000000000000004\",\"0.4\",\"0.5\",\"0.6000000000000001\",\"0.7000000000000001\",\"0.8\",\"0.9\",\"1.0\",\"1.1\",\"1.2000000000000002\",\"1.3\",\"1.4000000000000001\",\"1.5\",\"1.6\",\"1.7000000000000002\",\"1.8\",\"1.9000000000000001\",\"2.0\",\"2.1\",\"2.2\",\"2.3000000000000003\",\"2.4000000000000004\",\"2.5\",\"2.6\",\"2.7\",\"2.8000000000000003\",\"2.9000000000000004\"],\n",
+       "y: [\"-3.0\",\"-2.8\",\"-2.6\",\"-2.4\",\"-2.2\",\"-2.0\",\"-1.7999999999999998\",\"-1.5999999999999999\",\"-1.4\",\"-1.2\",\"-1.0\",\"-0.7999999999999998\",\"-0.5999999999999996\",\"-0.3999999999999999\",\"-0.19999999999999973\",\"0.0\",\"0.20000000000000018\",\"0.40000000000000036\",\"0.6000000000000001\",\"0.8000000000000003\",\"1.0\",\"1.2000000000000002\",\"1.4000000000000004\",\"1.6000000000000005\",\"1.8000000000000007\",\"2.0\",\"2.2\",\"2.4000000000000004\",\"2.6000000000000005\",\"2.8000000000000007\"],\n",
+       "mode: 'lines',\n",
+       "xaxis: 'x',\n",
+       "yaxis: 'y',\n",
+       "type: 'scatter',\n",
+       "name: 'Tangent line(x=1)',\n",
+       "};\n",
+       "\n",
+       "\n",
+       "var data = [ trace0, trace1];\n",
+       "Plotly.newPlot(target_b061aabd083f4daeac81c7044e2bcf25, data, layout);\n",
+       "})</script>\n"
+      ],
+      "text/plain": [
+       "tech.tablesaw.plotly.components.Figure@a73afdb"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import ai.djl.ndarray.types.DataType;\n",
+    "NDArray X = manager.arange(0f, 3f, 0.1f, DataType.FLOAT64);\n",
+    "double[] x = X.toDoubleArray();\n",
+    "\n",
+    "double[] fx = new double[x.length];\n",
+    "for (int i=0; i < x.length; i++) fx[i] = f.apply(x[i]);\n",
```

Comment:
Can you add the curly braces to this block to match the DJL convention (we usually verify this with checkstyle but it doesn't work on notebooks)?

Reply:
You mean only to the for loop right?

Reply:
Right, to the for loops (there are two of them here)

Reply:
Alright, done.